### ISSUE TITLE
appium 1.6.4: [XCUITest] Error: Need HOME env var to be set in order to adjust WDA attachments permission

### ISSUE THREAD

**Issue by HoonHee** [appium/appium#8256]
## Environment

* Appium 1.6.4
* macOS 10.12.3
* Node.js v7.8.0, npm v4.2.0
* iOS 10.3.1, iPhone 7 Plus
* Real device 

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium] Welcome to Appium v1.6.4

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium] Non-default server args:

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]   port: 4724

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]   backendRetries: 1

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]   sessionOverride: true

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]   defaultCapabilities: {

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]     udid: xxxxxxxxx

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]     automationName: XCUITest

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]     wdaLocalPort: 8100
[Appium]     fullReset: true

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]   }

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium] Default capabilities, which will be added to each request unless overridden by desired capabilities:

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]   udid: 'xxxxxxxxxx'

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]   automationName: 'XCUITest'

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]   wdaLocalPort: 8100

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium]   fullReset: true

170414124534 : [MANAGER_WARN] : AppiumLog(1): [Appium] Appium REST http interface listener started on 0.0.0.0:4724

170414124534 : [MANAGER_WARN] : [#### Info ####] End!! Appium Connect!! -- successed !! -- device No -- 1
170414124534 : [MANAGER_WARN] : ###############################################################

170414124544 : [MANAGER_INFO] : ================packet 6 ==== 1 =============
170414124544 : [MANAGER_WARN] : -[CommunicatorWithDC __processInstallCommandByPacket:deviceNo:], 1
170414124544 : [MANAGER_INFO] : ===== CMD_INSTALL Safari  ============ 1
170414124544 : [MANAGER_WARN] : path = Safari
170414124544 : [MANAGER_WARN] : -[ConnectionItemInfo finishResourceMornitor], 1
170414124544 : [MANAGER_WARN] : -[ConnectionItemInfo setupWebDriverWithBundleId], 1
170414124544 : [MANAGER_WARN] : AppiumLog(1): [HTTP] --> GET /wd/hub/status {}

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [MJSONWP] Calling AppiumDriver.getStatus() with args: []

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.6.4","revision":null}}

170414124544 : [MANAGER_WARN] : AppiumLog(1): [HTTP] <-- GET /wd/hub/status 200 47 ms - 83 

170414124544 : [MANAGER_WARN] : -[ConnectionItemInfo setupWebDriverWithBundleId] -- bundleId com.onycom.ResourceMornitor2
170414124544 : [MANAGER_WARN] : AppiumLog(1): [HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"platformName":"iOS","launchTimeout":600000,"deviceName":"iPhone7plus","newCommandTimeout":0,"platformVersion":"10.3.1","udid":"xxxxxxxx","bundleId":"com.onycom.ResourceMornitor2","fullReset":false,"automationName":"XCUITest","useNewWDA":true,"wdaLocalPort":8100}}

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [MJSONWP] Calling AppiumDriver.createSession() with args: [{"platformName":"iOS","launchTimeout":600000,"deviceName":"iPhone7plus","newCommandTimeout":0,"platformVersion":"10.3.1","udid":"xxxxxxxx","bundleId":"com.onycom.ResourceMornitor2","fullReset":false,"automationName":"XCUITest","useNewWDA":true,"wdaLocalPort":8100},null,null,null,null]

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'newSessionRequested' logged at 1492141544626 (12:45:44 GMT+0900 (KST))

170414124544 : [MANAGER_WARN] : AppiumLog(1): [Appium] Creating new XCUITestDriver (v2.27.2) session

170414124544 : [MANAGER_WARN] : AppiumLog(1): [Appium] Capabilities:

170414124544 : [MANAGER_WARN] : AppiumLog(1): [Appium]   platformName: 'iOS'

170414124544 : [MANAGER_WARN] : AppiumLog(1): [Appium]   launchTimeout: 600000

170414124544 : [MANAGER_WARN] : AppiumLog(1): [Appium]   deviceName: 'iPhone7plus'
[Appium]   newCommandTimeout: 0
[Appium]   platformVersion: '10.3.1'
[Appium]   udid: 'xxxxxxxx'
[Appium]   bundleId: 'com.onycom.ResourceMornitor2'

170414124544 : [MANAGER_WARN] : AppiumLog(1): [Appium]   fullReset: false
[Appium]   automationName: 'XCUITest'

170414124544 : [MANAGER_WARN] : AppiumLog(1): [Appium]   useNewWDA: true

170414124544 : [MANAGER_WARN] : AppiumLog(1): [Appium]   wdaLocalPort: 8100

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] XCUITestDriver version: 2.27.2

170414124544 : [MANAGER_WARN] : AppiumLog(1): [BaseDriver] Session created with session id: 39cae946-a3e9-408a-ba34-c467fc8e3fef

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Xcode version set to '8.3.1' (tools v8.2.0.0.1.1480973914)

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] iOS SDK Version set to '10.3'

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'xcodeDetailsRetrieved' logged at 1492141544802 (12:45:44 GMT+0900 (KST))

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Available devices: xxxxxxxx

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Creating iDevice object with udid 'xxxxxx'

170414124544 : [MANAGER_WARN] : AppiumLog(1): [XCUITest] Determining device to run tests on: udid: 'xxxxxx', real device: true

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'appConfigured' logged at 1492141544892 (12:45:44 GMT+0900 (KST))
[debug] [BaseDriver] Event 'resetStarted' logged at 1492141544893 (12:45:44 GMT+0900 (KST))

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Reset: running ios real device reset flow

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'resetComplete' logged at 1492141544897 (12:45:44 GMT+0900 (KST))

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [iOSLog] Attempting iOS device log capture via libimobiledevice idevicesyslog

170414124544 : [MANAGER_WARN] : AppiumLog(1): [debug] [iOSLog] Found idevicesyslog: '/usr/local/bin/idevicesyslog'

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'logCaptureStarted' logged at 1492141545012 (12:45:45 GMT+0900 (KST))

170414124545 : [MANAGER_WARN] : AppiumLog(1): [XCUITest] Setting up real device

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'wdaStartAttempted' logged at 1492141545015 (12:45:45 GMT+0900 (KST))

170414124545 : [MANAGER_WARN] : AppiumLog(1): [XCUITest] Using WDA path: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'

170414124545 : [MANAGER_WARN] : AppiumLog(1): [XCUITest] Using WDA agent: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj'

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Capability 'useNewWDA' set, so uninstalling WDA before proceeding

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Removing WDA application from device

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'wdaUninstalled' logged at 1492141545275 (12:45:45 GMT+0900 (KST))

170414124545 : [MANAGER_WARN] : AppiumLog(1): [XCUITest] Launching WebDriverAgent on the device

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Carthage found: '/usr/local/bin/carthage'

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Killing hanging processes

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Beginning test with command 'xcodebuild build-for-testing test-without-building -project /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=xxxxxxxx -configuration Debug IPHONEOS_DEPLOYMENT_TARGET=10.3' in directory '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Output from xcodebuild will not be logged

170414124545 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Starting iproxy to forward traffic from local port 8100 to device port 8100 over USB

170414124550 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Waiting up to 60000ms for WebDriverAgent to start
[debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124551 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124551 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Log file for xcodebuild test: /Users/onycom_test/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/E5DD5458-535B-4241-9408-1D4CF216BCD2/Session-WebDriverAgentRunner-2017-04-14_124551-hZ1DVI.log

170414124551 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124552 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124552 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124553 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124553 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124554 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124554 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124555 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124555 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124556 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124557 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124557 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124558 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124558 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124558 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"state\" : \"success\",\n    \"os\" : {\n      \"name\" : \"iOS\",\n      \"version\" : \"10.2.1\"\n    },\n    \"ios\" : {\n      \"simulatorVersion\" : \"10.2.1\",\n      \"ip\" : \"192.168.0.13\"\n    },\n    \"build\" : {\n      \"time\" : \"Apr 14 2017 12:36:52\"\n    }\n  },\n  \"sessionId\" : \"B4951665-CEF4-45AA-8BE9-6AEB3720D9C4\",\n  \"status\" : 0\n}"

170414124558 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] WebDriverAgent running on ip '192.168.0.13'

170414124558 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] WebDriverAgent successfully started after 8155ms

170414124558 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1492141558553 (12:45:58 GMT+0900 (KST))
[debug] [XCUITest] Sending createSession command to WDA

170414124558 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.onycom.ResourceMornitor2","arguments":[],"environment":{},"shouldWaitForQuiescence":true,"shouldUseTestManagerForVisibilityDetection":false,"maxTypingFrequency":60,"shouldUseSingletonTestManager":true}}

170414124601 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Got response with status 200: {"value":{"sessionId":"46B41919-D204-4928-A016-72CE8964262C","capabilities":{"device":"iphone","browserName":"ClientResourceMornitor","sdkVersion":"10.2.1","CFBundleIdentifier":"com.onycom.ResourceMornitor2"}},"sessionId":"46B41919-D204-4928-A016-72CE8964262C","status":0}

170414124601 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'wdaSessionStarted' logged at 1492141561446 (12:46:01 GMT+0900 (KST))

170414124611 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'wdaStartAttempted' logged at 1492141571450 (12:46:11 GMT+0900 (KST))

170414124611 : [MANAGER_WARN] : AppiumLog(1): [XCUITest] Using WDA path: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'

170414124611 : [MANAGER_WARN] : AppiumLog(1): [XCUITest] Using WDA agent: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj'

170414124611 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Capability 'useNewWDA' set, so uninstalling WDA before proceeding

170414124611 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Removing WDA application from device

170414124611 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'wdaUninstalled' logged at 1492141571754 (12:46:11 GMT+0900 (KST))
[XCUITest] Launching WebDriverAgent on the device

170414124611 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Carthage found: '/usr/local/bin/carthage'
[debug] [XCUITest] Killing hanging processes

170414124611 : [MANAGER_WARN] : AppiumLog(1): [XCUITest] xcodebuild exited with code 'null' and signal 'SIGKILL'

170414124611 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] iproxy exited with code 'null'

170414124611 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Beginning test with command 'xcodebuild build-for-testing test-without-building -project /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=xxxxxxxx -configuration Debug IPHONEOS_DEPLOYMENT_TARGET=10.3' in directory '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'
[debug] [XCUITest] Output from xcodebuild will not be logged

170414124611 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Starting iproxy to forward traffic from local port 8100 to device port 8100 over USB

170414124617 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Waiting up to 60000ms for WebDriverAgent to start
[debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124617 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124618 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Log file for xcodebuild test: /Users/onycom_test/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/AB1C4DFE-184B-4D7F-9C4A-F36102C1BEE1/Session-WebDriverAgentRunner-2017-04-14_124618-CTtzz7.log

170414124618 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124618 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124619 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124619 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124620 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124620 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124621 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124621 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124622 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124623 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124623 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124624 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124624 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124625 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body

170414124625 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Got response with status 200: "{\n  \"value\" : {\n    \"state\" : \"success\",\n    \"os\" : {\n      \"name\" : \"iOS\",\n      \"version\" : \"10.2.1\"\n    },\n    \"ios\" : {\n      \"simulatorVersion\" : \"10.2.1\",\n      \"ip\" : \"192.168.0.13\"\n    },\n    \"build\" : {\n      \"time\" : \"Apr 14 2017 12:36:52\"\n    }\n  },\n  \"sessionId\" : \"2A645AD6-553E-4EE1-A29A-75945C2717F5\",\n  \"status\" : 0\n}"

170414124625 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] WebDriverAgent running on ip '192.168.0.13'

170414124625 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] WebDriverAgent successfully started after 8151ms

170414124625 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'wdaSessionAttempted' logged at 1492141585062 (12:46:25 GMT+0900 (KST))

170414124625 : [MANAGER_WARN] : AppiumLog(1): [debug] [XCUITest] Sending createSession command to WDA

170414124625 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"com.onycom.ResourceMornitor2","arguments":[],"environment":{},"shouldWaitForQuiescence":true,"shouldUseTestManagerForVisibilityDetection":false,"maxTypingFrequency":60,"shouldUseSingletonTestManager":true}}

170414124627 : [MANAGER_WARN] : AppiumLog(1): [debug] [JSONWP Proxy] Got response with status 200: {"value":{"sessionId":"82B2C7B6-A6B0-4196-BAB6-4D228D497EE0","capabilities":{"device":"iphone","browserName":"ClientResourceMornitor","sdkVersion":"10.2.1","CFBundleIdentifier":"com.onycom.ResourceMornitor2"}},"sessionId":"82B2C7B6-A6B0-4196-BAB6-4D228D497EE0","status":0}

170414124627 : [MANAGER_WARN] : AppiumLog(1): [debug] [BaseDriver] Event 'wdaSessionStarted' logged at 1492141587918 (12:46:27 GMT+0900 (KST))

170414124637 : [MANAGER_WARN] : 

[#### Info ####] availableNotification -- fail !!

170414124637 : [MANAGER_WARN] : AppiumLog(1): error task - [XCUITest] Error: Need HOME env var to be set in order to adjust WDA attachments permission
    at adjustWDAAttachmentsPermissions$ (../../lib/utils.js:91:11)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at enqueueResult (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:185:17)
    at F (/usr/local/lib/node_modules/appium/node_modules/core-js/library/modules/$.export.js:30:36)
    at AsyncIterator.enqueue (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:184:12)
    at AsyncIterator.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at Object.runtime.async (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:209:12)
    at adjustWDAAttachmentsPermissions (../../lib/utils.js:98:26)
    at XCUITestDriver.callee$2$0$ (../../lib/driver.js:411:13)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
 Error: Need HOME env var to be set in order to adjust WDA attachments permission
    at adjustWDAAttachmentsPermissions$ (../../lib/utils.js:91:11)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at enqueueResult (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:185:17)
    at F (/usr/local/lib/node_modules/appium/node_modules/core-js/library/modules/$.export.js:30:36)
    at AsyncIterator.enqueue (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:184:12)
    at AsyncIterator.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at Object.runtime.async (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:209:12)
    at adjustWDAAttachmentsPermissions (../../lib/utils.js:98:26)
    at XCUITestDriver.callee$2$0$ (../../lib/driver.js:411:13)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)

170414124637 : [MANAGER_WARN] : [#### Info ####] End!! Appium Connect!! -- Failed !! -- device No -- 1
170414124637 : [MANAGER_WARN] : ###############################################################

170414124637 : [MANAGER_WARN] : AppiumLog(1): error task - [MJSONWP] Encountered internal error running command: Error: Need HOME env var to be set in order to adjust WDA attachments permission
    at adjustWDAAttachmentsPermissions$ (../../lib/utils.js:91:11)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at enqueueResult (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:185:17)
    at F (/usr/local/lib/node_modules/appium/node_modules/core-js/library/modules/$.export.js:30:36)
    at AsyncIterator.enqueue (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:184:12)
    at AsyncIterator.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at Object.runtime.async (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:209:12)
    at adjustWDAAttachmentsPermissions (../../lib/utils.js:98:26)
    at XCUITestDriver.deleteSession$ (../../lib/driver.js:442:13)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)

170414124637 : [MANAGER_WARN] : [#### Info ####] End!! Appium Connect!! -- Failed !! -- device No -- 1
170414124637 : [MANAGER_WARN] : ###############################################################

**Comment by mykola-mokhnach** [comment]
Make sure all the necessary environment variables are properly set if you start Appium as a daemon on system startup.

**Comment by mykola-mokhnach** [comment]
closed as duplicate of https://github.com/appium/appium/issues/7663

**Comment by HoonHee** [comment]
i set environment variable.

    /usr/bin
    /bin
    /usr/sbin
    /sbin
    /usr/local/bin

**Comment by mykola-mokhnach** [comment]
http://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html

**Comment by HoonHee** [comment]
thanks for reply

my program execute Appium 1.6.4 by NSTask.

NSTask * appiumTask = [[NSTask alloc] init];
appiumTask.environment = [NSDictionary dictionaryWithObjectsAndKeys:@"/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin", @"PATH", nil];

appium1.5.13 is working.
appium 1.6.0 ~ 3 is working.
but appium 1.6.4. is not working.

What environment variables should I add?

**Comment by mykola-mokhnach** [comment]
Copypaste from http://stackoverflow.com/questions/386783/nstask-not-picking-up-path-from-the-users-environment

```objc
   [task setLaunchPath:@"/bin/bash"];
    NSArray *args = [NSArray arrayWithObjects:@"-l",
                     @"-c",
                     @"my command",
                     nil];
    [task setArguments: args];
```

**Comment by HoonHee** [comment]
thanks for reply.

I understand the concept of environment variables and how to run NSTask.
I ran Appium using NStask in my program.
From Appium 1.5.13 to Appium 1.6.3 we ran all the tests and tested the iPhone.

However, Appium only updates to 1.6.4 in the same program and is not running.
Appium's log shows "appium 1.6.4: [XCUITest] Error: Need HOME env var to be set in order to adjust".

I say you have an environment variable problem, but I do not know what value to add.

Using Google Translator
